packages/subgraph/src/utils/tokens/erc20.ts

```diff
@@ -0,0 +1,217 @@
+import {Address, BigInt, Bytes, ethereum, log} from '@graphprotocol/graph-ts';
+import {
+  ERC20Balance,
+  ERC20Contract,
+  ERC20Transfer
+} from '../../../generated/schema';
+import {ERC20} from '../../../generated/templates/DaoTemplate/ERC20';
+import {
+  ERC20_transfer,
+  ERC20_transferFrom,
+  getTransferId,
+  TransferType
+} from './common';
+
+export function fetchERC20(address: Address): ERC20Contract | null {
+  let erc20 = ERC20.bind(address);
+
+  // Try load entry
+  let contract = ERC20Contract.load(address.toHexString());
+  if (contract != null) {
+    return contract;
+  }
+
+  contract = new ERC20Contract(address.toHexString());
+
+  let try_name = erc20.try_name();
+  let try_symbol = erc20.try_symbol();
+  let totalSupply = erc20.try_totalSupply();
+
+  if (totalSupply.reverted) {
+    return null;
+  }
+  contract.name = try_name.reverted ? '' : try_name.value;
+  contract.symbol = try_symbol.reverted ? '' : try_symbol.value;
+
+  contract.save();
+
+  return contract;
+}
+
+export function updateERC20Balance(
+  token: Address,
+  dao: Address,
+  amount: BigInt,
+  timestamp: BigInt,
+  type: TransferType
+): void {
+  let daoId = dao.toHexString();
+  let balanceId = daoId.concat('_').concat(token.toHexString());
+
+  let erc20Balance = ERC20Balance.load(balanceId);
+
+  if (!erc20Balance) {
+    erc20Balance = new ERC20Balance(balanceId);
+    erc20Balance.dao = daoId;
+    erc20Balance.token = token.toHexString();
+    erc20Balance.balance = BigInt.zero();
+  }
+
+  let erc20 = ERC20.bind(token);
+  let balance = erc20.try_balanceOf(dao);
+  if (!balance.reverted) {
+    erc20Balance.balance = balance.value;
+  }
+
+  erc20Balance.lastUpdated = timestamp;
+  erc20Balance.save();
+}
+
+export function handleERC20Action(
+  token: Address,
+  dao: Address,
+  proposalId: string,
+  data: Bytes,
+  actionIndex: number,
+  event: ethereum.Event
+): void {
+  let contract = fetchERC20(token);
+  if (!contract) {
+    return;
+  }
+
+  let decodeABI = '';
+
+  let functionSelector = data.toHexString().substring(0, 10);
+  let calldata = data.toHexString().slice(10);
+
+  if (functionSelector == ERC20_transfer) {
+    decodeABI = '(address,uint256)';
+  }
+
+  if (functionSelector == ERC20_transferFrom) {
+    decodeABI = '(address,address,uint256)';
+  }
+
+  let decoded = ethereum.decode(decodeABI, Bytes.fromHexString(calldata));
+
+  if (!decoded) {
+    return;
+  }
+
+  let tuple = decoded.toTuple();
+
+  let from = new Address(0);
+  let to = new Address(0);
+  let amount = BigInt.zero();
+
+  if (functionSelector == ERC20_transfer) {
+    from = dao;
+    to = tuple[0].toAddress();
+    amount = tuple[1].toBigInt();
+  }
+
+  if (functionSelector == ERC20_transferFrom) {
+    from = tuple[0].toAddress();
+    to = tuple[1].toAddress();
+    amount = tuple[2].toBigInt();
+  }
+
+  // Ambiguity ! No need to store transfer such as this.s
+  if (from == to) {
```

Comment:
Why is this ambiguous? 

`from == to` makes no sense ofc, but I am not sure if we shouldn't handle these edge cases differently than simply returning.

Isn't this, formally, still a tx / a transfer that we should index?